### PR TITLE
Fixup PPI by refreshing the tree on replacement

### PR DESCRIPTION
**Pull request by atoomic**
Avoid __insert_before_child PPI issues

Add some extra tests and provides a fixup for them.

### CODE REVIEW

**Comment by atoomic** [comment]
https://travis-ci.org/atoomic/ref-util-rewriter/builds/526960698

File: lib/Ref/Util/Rewriter.pm [hunk 1]
```diff
@@ -171,6 +169,11 @@ sub rewrite_doc {
             }
 
             $ref_sub->remove;
+
+            # update statements... to avoid PPI issues when moving elements...
+            # this is very ugly... but probably the best solution
+            $all_statements = $doc->find('PPI::Statement');
+            goto ALL_STATEMENTS;
```
Comment (xsawyerx) [inline lib/Ref/Util/Rewriter.pm:176]: I don't like this either, but I couldn't think of a better way to do it at the moment. I moved it to a function with a `while` but it didn't work well. But I would really appreciate coming up with a better way to do this.

**Comment by xsawyerx** [comment]
Merged, thanks! :+1:

**Comment by xsawyerx** [comment]
It doesn't show "Merged" by GH because I cherry-picked only the relevant commits.

**Pull request closed without merging**